src/components/text/index.tsx

```diff
@@ -62,47 +65,90 @@ class Text extends PureComponent<PropsTypes> {
   //   this._root.setNativeProps(nativeProps); // eslint-disable-line
   // }
 
-  getTextPartsByHighlight(targetString = '', highlightString = '') {
-    if (_.isEmpty(highlightString.trim())) {
-      return [targetString];
+  getPartsByHighlight(targetString = '', highlightString: string | string[]) {
+    if (typeof highlightString === 'string') {
+      if (_.isEmpty(highlightString.trim())) {
+        return [{string: targetString, shouldHighlight: false}];
+      }
+      return this.getTextPartsByHighlight(targetString, highlightString);
+    } else {
+      return this.getArrayPartsByHighlight(targetString, highlightString);
     }
+  }
 
+  getTextPartsByHighlight(targetString = '', highlightString = '') {
+    if (highlightString === '') {
+      return [{string: targetString, shouldHighlight: false}];
+    }
     const textParts = [];
     let highlightIndex;
-
     do {
       highlightIndex = targetString.toLowerCase().indexOf(highlightString.toLowerCase());
       if (highlightIndex !== -1) {
         if (highlightIndex > 0) {
-          textParts.push(targetString.substring(0, highlightIndex));
+          textParts.push({string: targetString.substring(0, highlightIndex), shouldHighlight: false});
         }
-        textParts.push(targetString.substr(highlightIndex, highlightString.length));
+        textParts.push({string: targetString.substr(highlightIndex, highlightString.length), shouldHighlight: true});
         targetString = targetString.substr(highlightIndex + highlightString.length);
       } else {
-        textParts.push(targetString);
+        textParts.push({string: targetString, shouldHighlight: false});
       }
     } while (highlightIndex !== -1);
 
     return textParts;
   }
 
+  getArrayPartsByHighlight(targetString = '', highlightString = ['']) {
+    const target = _.toLower(targetString);
+    const indices = [];
+    let index = 0;
+    let lastWordLength = 0;
+    for (let j = 0; j < highlightString.length; j++) {
+      const word = _.toLower(highlightString[j]);
+      const targetSuffix = target.substring(index + lastWordLength);
+      const i = targetSuffix.indexOf(word);
+      if (i >= 0) {
+        const newIndex = index + lastWordLength + i;
+        indices.push(index + lastWordLength + i);
+        indices.push(index + lastWordLength + i + word.length);
+        index = newIndex;
+        lastWordLength = word.length;
+      } else {
+        break;
+      }
+    }
```

Comment:
I have a suggestion that can shorten this code a little (: 
Let me know if it can work.. 

First, regardless.. lodash has lots of great utils I'd use instead of for..loops when possible, I find them easier to read. 

Anyway, there's the `_.split` util.
You can go over the target string and split it according to the highlight strings array and this way break the strings as you did without having to save their indices. 

Here I had to understand you saved the indices in couples of start and end of highlight and it's not really clear at first glance.

Reply:
I can't find a way to make it work with the split method.
If I don't keep the indices I can only return a lower case string and not the original one...

Reply:
Yes, I understand. 

It's just I'd like to make the code a little more readable. 
(Btw, my old code there is not so readable also)

Everything together makes the Text component to look a bit too much so I was trying to think of ways to clean this up a little before it all become legacy code. 

...Hmm another thought of how to shorten this up. 
Do you think we can take the use case of a **single highlight** string and use your implementation as well? 
(and then we can remove my old implementation)

Technically if the user pass a string (and not an array of strings) we can just pass it to your method wrapped with an array. 
WDYT?

Reply:
Actually, we said that if the user passes only one string we highlight all the repeats of this string (as the old method does), and if he passes an array of strings we highlight it according to its order (as the new method does)

Reply:
Oh right, I remember there was a good reason for the separation.